doc/adr/0003-D3-charts.md

```diff
@@ -0,0 +1,160 @@
+	.domain([this.state.xyData.labels[0], this.state.xyData.labels[xDomain]].map(Number))
+	.range([0, chartWidth]);
+
+const xScale2 = scaleLinear()
+	.domain([this.state.xyData.labels[0], this.state.xyData.labels[xDomain]].map(Number))
+	.range([this.margin.left, chartWidth - this.margin.right]);
+
+const lineData = line()
+	.x((d: any) => xScale(Number(d.xValue)))
+	.y((d: any) => yScale(+d.yValue) - 0.5)
+	.context(ctx);
+```
+
+- Drawing the chart
+
+```js
+ctx.clearRect(0, 0, chartWidth, chartHeight);
+ctx.save();
+data.forEach((row, i) => {
+	ctx.lineWidth = 2;
+	ctx.strokeStyle = this.state.xyData.datasets[i].borderColor;
+	ctx.fillStyle = this.state.xyData.datasets[i].borderColor;
+	if (this.isScatterPlot){
+		row.forEach((point: any)=> {
+			ctx.beginPath();
+			const cx = xScale2(Number(start) + point.xValue);
+			const cy = yScale(point.yValue);
+			ctx.arc(cx, cy, 2, 0, 2 * Math.PI);
+			ctx.stroke();
+			ctx.fill();
+			ctx.closePath();
+		});
+	} else {
+		ctx.beginPath();
+		lineData(row);
+		ctx.stroke();
+		ctx.closePath();
+	}
+});
+ctx.restore();
+```
+- Adjust references and remove ChartJS
+
+Replace occurrences of `this.chartRef.current.chartInstance` with `this.chartRef.current`.
+
+A [full implementation][implementation] of this solution is available as part of this ADR.
+
+For comparison, here are the line chart and the scatter chart rendered using D3:
+
+![Examples with D3](0003/Example_D3.png)
+
+And the same charts rendered with ChartJS:
+
+![Examples with ChartJS](0003/Example_ChartJS.png)
+
+## Consequences
+
+### What becomes more difficult
+
+D3 has a steep learning curve. Even the most simple chart requires deep understanding of how many libraries and manipulation of the data so that it can be used by these libraries.
+
+Every different chart requires again time and practice to be implemented.
+
+Specially the first implementation of a new solution with D3 takes a long time of study and trial in order to work, more so than libraries like ChartJS.
+
+### What becomes easier
+
+After a first example is implemented, like a line chart, the same code can be easily reused for similar solutions. Therefore, having a team member with experience in D3 greatly reduces the effort necessary to learn and use the library.
+
+Due to its versatility, nearly any solution can be implemented in the project. The product is not limited by the few solutions of a library, as it happens with ChartJS.
+
+Also, customization becomes more accessible. As stated above, some aspects of the tooltip could not be implemented in this project due to the limitations of ChartJS.
+
+### Conclusion
+
+D3 is a complex and powerful library with a steep learning curve. But having a team member with experience using it greatly diminishes the barriers of adoption.
+
+Once it is integrated into the project, great possibilities for customization and expansion of functionality become accessible.
+
+[implementation]: ./0003/proto_xy-output-component.tsx
```

Comment:
This whole file lends 42 markdownlint warnings in VS Code, locally. Would fix all those before merging (a DoD policy of ours).

Reply:
Warnings fixed with the addition of the `.markdownrc` file to the root of the project, which shall normalise the use of the markdownlint plugin for everybody.

Reply:
File name is `.markdownlintrc` though, and when I open it alongside this ADR `.md` file, I still get 40 warnings in VS Code. Are we missing some setup step for it, to maybe document?

Reply:
Ah, the markdownlint plugin for VS Code doesn't recognise rc files. I changed it to a js file. Now the warnings should not appear anymore.

Reply:
The warnings still show, and this `.markdownlint.js` file generates syntax errors in VS Code. It is unlikely to be a JS file indeed. Its name also starts with a dot prefix, which is more for non-JS or real "dot" (here, RC) files.

Reply:
I was following their instructions from https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint:

> "Rules can be enabled, disabled, and customized by creating a JSON file named .markdownlint.jsonc/.markdownlint.json or a YAML file named .markdownlint.yaml/.markdownlint.yml **or a JavaScript file named .markdownlint.js** in any directory of a project."

I guess it doesn't work with `.js` either. I changed to `.json` and tested locally on my VS Code.